src/Components/WebAssembly/Server/src/ComponentsWebAssemblyApplicationBuilderExtensions.cs

```diff
@@ -16,74 +15,50 @@ namespace Microsoft.AspNetCore.Builder
     /// <summary>
     /// Extensions for mapping Blazor WebAssembly applications.
     /// </summary>
-    public static class ComponentsWebAssemblyEndpointRouteBuilderExtensions
+    public static class ComponentsWebAssemblyApplicationBuilderExtensions
     {
         /// <summary>
         /// Maps a Blazor webassembly application to the <paramref name="pathPrefix"/>.
         /// </summary>
-        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="builder">The <see cref="IApplicationBuilder"/>.</param>
         /// <param name="pathPrefix">The <see cref="PathString"/> that indicates the prefix for the Blazor application.</param>
-        /// <returns>The <see cref="IEndpointConventionBuilder"/></returns>
-        public static IEndpointConventionBuilder MapBlazorWebAssemblyApplication(this IEndpointRouteBuilder endpoints, PathString pathPrefix)
+        /// <returns>The <see cref="IApplicationBuilder"/></returns>
+        public static IApplicationBuilder MapBlazorFrameworkFiles(this IApplicationBuilder builder, PathString pathPrefix)
         {
-            if (endpoints is null)
+            if (builder is null)
             {
-                throw new ArgumentNullException(nameof(endpoints));
+                throw new ArgumentNullException(nameof(builder));
             }
 
-            var webHostEnvironment = endpoints.ServiceProvider.GetRequiredService<IWebHostEnvironment>();
+            var webHostEnvironment = builder.ApplicationServices.GetRequiredService<IWebHostEnvironment>();
 
             var options = CreateStaticFilesOptions(webHostEnvironment.WebRootFileProvider);
-            var appBuilder = endpoints.CreateApplicationBuilder();
 
-            appBuilder.Use(async (ctx, next) =>
+            builder.MapWhen(ctx => ctx.Request.Path.StartsWithSegments(pathPrefix, out var rest) && rest.StartsWithSegments("/_framework") && !rest.StartsWithSegments("/_framework/blazor.server.js"),
```

Comment:
Not that I'm greatly concerned about this, but why is it worth excluding `blazor.server.js` specifically? Is this anticipating a case where someone has Blazor WebAssembly and Blazor Server applications both running from the same base URL?

Reply:
Yep, like our tests :)

Reply:
Our E2E tests do this, and its a good exception to have here. I plan to improve this in preview3.